### PR TITLE
Feature/43 unit test

### PR DESCRIPTION
**Pull request by JamieYuu**
Added unit test feature to the project

### CODE REVIEW

File: server/app/unit_test/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+python -m unittest test_cases
+```
+
+It will take a few seconds to run the test cases, you will see the following message if all test cases pass:
+```
+----------------------------------------------------------------------
+Ran 8 tests in 1.967s
+
+OK
+```
```
Comment (nirlipo) [inline server/app/unit_test/README.md]: When it runs, can it give you some output messages of the tests it passed?

Why didn't we use django.tests?

File: server/app/unit_test/test_cases.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        self.assertFalse('Foo'.isupper())
+
+    def test_split(self):
+        s = 'hello world'
+        self.assertEqual(s.split(), ['hello', 'world'])
+        # check that s.split fails when the separator is not a string
+        with self.assertRaises(TypeError):
+            s.split(2)
+
+    def test_visitall(self):
```
Comment (nirlipo) [inline server/app/unit_test/test_cases.py]: add comments to the functions please. What is being tested in visitall?

File: server/app/unit_test/test_cases.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+        with open(os.path.abspath(os.path.dirname(__file__) + '/../unit_test/' + "test.vfg"), "w") as f:
+            json.dump(final, f)
+
+        output_content = open(os.path.abspath(os.path.dirname(__file__) + '/../unit_test/' + "test.vfg"), 'r', encoding='utf-8-sig').read()
+
+        expected_content = open(os.path.abspath(os.path.dirname(__file__) + '/../unit_test/' + "expect_output.vfg"), 'r', encoding='utf-8-sig').read()
+        #print(output_content, expected_content)
+
+        self.assertTrue(filecmp.cmp(os.path.abspath(os.path.dirname(__file__) + '/../unit_test/' + 'test.vfg'), os.path.abspath(os.path.dirname(__file__) + '/../unit_test/' + 'expect_output.vfg'), shallow=False))
```
Comment (nirlipo) [inline server/app/unit_test/test_cases.py]: Shouldn't be an error message string that gives information about which part of the test failed? see this example: https://github.com/cgoldberg/python-unittest-tutorial

**Review by nirlipo — CHANGES_REQUESTED** [review]
Thanks for the PR!

Besides the comments in the files, I have 2 more questions:

 - Should we move the test folder out of app?
 - Can you group the input files into different folders?

Thanks!

**Comment by JamieYuu** [comment]
Hi @nirlipo , sorry for the late response, I was focusing on the planning-as-service project and forgot to check the messages here. I've made a few changes to the unit test branch according to your comments. 

For the django.test, we found the document says
> Django’s unit tests use a Python standard library module: unittest

From a functional point of view, using Django.test is the same as using the Python unittest module directly. So we think just using the Python unittest module would be more light and easy to use, and won't cause any potential conflict with the existing features.
We are looking forward to your reply, thank you for your time!

File: server/unit_test/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+The unit tests can be run with the command:
+```
+python -m unittest test_cases
+```
+
+It will take a few seconds to run the test cases, you will see the following message if all test cases pass:
+```
+----------------------------------------------------------------------
+Ran 8 tests in 1.967s
```
Comment (nirlipo) [inline server/unit_test/README.md:12]: Can you update the output of the new test_cases.py run?

**Review by nirlipo — CHANGES_REQUESTED** [review]
Thanks for the changes. Please update the output in the Readme and we are ready to merge!

**Review by nirlipo — APPROVED** [review]
Thanks!

**Comment by nirlipo** [comment]
Please let me know how to include this on the travis pipeline!

**Pull request merged**